qiskit_nature/problems/second_quantization/lattice/lattice/triangular_lattice.py

```diff
@@ -0,0 +1,296 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The triangular lattice"""
+from dataclasses import asdict
+from itertools import product
+from math import pi
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+from retworkx import PyGraph
+
+from .lattice import DrawStyle, Lattice
+from .boundary_condition import BoundaryCondition
+
+
+def _coordinate_to_index(coord: np.ndarray, size: Tuple[int, int]) -> int:
+    """Convert the coordinate of a lattice point to an integer for labeling.
+        When size=(l0, l1), then a coordinate (x0, x1) is converted as
+        x0 + x1*l0.
+    Args:
+        coord: Input coordinate to be converted.
+        size: Lengths of each dimension.
+
+    Returns:
+        int: Return x0 + x1*l0 when coord=np.array([x0, x1]) and size=(l0, l1).
+    """
+    dim = 2
+    base = np.array([np.prod(size[:i]) for i in range(dim)], dtype=int)
+    return np.dot(coord, base).item()
+
+
+def _self_loops(size: Tuple[int, int], onsite_parameter: complex) -> List[Tuple[int, int, complex]]:
```

Comment:
The same as my previous comment applies to there functions.

Reply:
I have included them in the class `TriangularLattice`.